app/controllers/session_controller.rb

```diff
@@ -0,0 +1,20 @@
+class SessionController < ApplicationController
+  def create
+    begin
+      @user = User.from_omniauth(request.env['omniauth.auth'])
+      session[:user_id] = @user.id
+      flash[:success] = "Olá, #{@user.name}."
+    rescue
+      flash[:danger] = "Erro ao autenticar o usuário."
+    end
+    redirect_to root_path
+  end
+
+  def destroy
+    if current_user
+      session.delete(:user_id)
+      flash[:success] = 'Até logo!'
+    end
+    redirect_to root_path
+  end
+end
```

Comment:
We should have a path (or even a controller) for each of our providers. As we are only using Github, we can keep it this way. We should also store several providers uuid, so perhaps we should have another model to store user OAuth data. We can work more on that later.